cou/steps/vault.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions for prereq steps relating to vault."""
+import logging
+
+from cou.exceptions import ApplicationNotFound, VaultSealed
+from cou.utils.juju_utils import Model
+
+logger = logging.getLogger(__name__)
+
+
+async def check_vault_status(model: Model) -> None:
+    """Make sure vault is not in sealed status.
```

Comment:
Just some docs suggestions:

- maybe name it something a bit clearer, like `verify_vault_is_unsealed`, or `validate_vault_is_unsealed`:

```suggestion
async def verify_vault_is_unsealed(model: Model) -> None:
    """Verify vault is unsealed.
    
    Check vault status. If vault is sealed, raise VaultSealed.
```

Reply:
Updated.